contracts/proposal/dao-proposal-single/src/state.rs

```diff
@@ -55,6 +58,8 @@ pub struct Config {
     /// remain open until the DAO's treasury was large enough for it to be
     /// executed.
     pub close_proposal_on_execution_failure: bool,
+    /// Optional time delay on proposal execution
+    pub timelock: Option<Timelock>,
```

Comment:
```suggestion
    /// Optional time delay on proposal execution during which the configured vetoer can veto the proposal.
    pub timelock: Option<Timelock>,
```

my initial thought was this comment could be improved, but i think `timelock` is just quite vague in general. what if we just call it `veto`? and the new status enum variant can be `VetoTimelock` instead of `Timelock`.

i think in general the `timelock` naming scheme could use a refactor to `veto` since that's really what it is